semantickernel-experimental/src/main/java/com/microsoft/semantickernel/data/recorddefinition/VectorStoreRecordDefinition.java

```diff
@@ -50,6 +49,52 @@ public List<VectorStoreRecordField> getAllFields() {
         return fields;
     }
 
+    public List<VectorStoreRecordField> getNonVectorFields() {
+        List<VectorStoreRecordField> fields = new ArrayList<>();
+        fields.add(keyField);
+        fields.addAll(dataFields);
+        return fields;
+    }
+
+    private List<Field> getDeclaredFields(Class<?> recordClass, List<VectorStoreRecordField> fields,
+        String fieldType) {
+        List<Field> declaredFields = new ArrayList<>();
+        for (VectorStoreRecordField field : fields) {
+            try {
+                Field declaredField = recordClass.getDeclaredField(field.getName());
+                declaredFields.add(declaredField);
+            } catch (NoSuchFieldException e) {
+                throw new IllegalArgumentException(
+                    String.format("%s field not found in record class: %s", fieldType,
+                        field.getName()));
+            }
+        }
+        return declaredFields;
+    }
+
+    public Field getKeyDeclaredField(Class<?> recordClass) {
+        try {
+            return recordClass.getDeclaredField(keyField.getName());
+        } catch (NoSuchFieldException e) {
+            throw new IllegalArgumentException(
+                "Key field not found in record class: " + keyField.getName());
+        }
+    }
+
+    public List<Field> getDataDeclaredFields(Class<?> recordClass) {
+        return getDeclaredFields(
+            recordClass,
+            dataFields.stream().map(f -> (VectorStoreRecordField) f).collect(Collectors.toList()),
+            "Data");
```

Comment:
Are "Data" and "Vector" the only two field types allowed? If the field types are limited to a known set, then and enum might be a better choice here.